openmdao/core/driver.py

```diff
@@ -35,6 +36,18 @@ class Driver(object):
     recording_options['excludes'] :  list of strings('')
         Patterns for variables to exclude in recording (processed after includes).
 
+    debug_print['debug_print'] :  bool(False)
+        If True, print debugging information every Driver iteration.
+    debug_print['debug_print_desvars'] :  bool(False)
+        If True and the option debug_print is True, print design variables every Driver iteration.
+    debug_print['debug_print_nl_con'] :  bool(False)
+        If True and the option debug_print is True, print nonlinear constraints every
+        Driver iteration.
+    debug_print['debug_print_ln_con'] :  bool(False)
+        If True and the option debug_print is True, print linear constraints every Driver iteration.
+    debug_print['debug_print_objective'] :  bool(False)
+        If True and the option debug_print is True, print objectives every Driver iteration.
+
```

Comment:
seems like some redundancy in the naming here..  suggest a more concise: 
```
    debug['print'] :  bool(False)
    debug['print_desvars'] :  bool(False)
    debug['print_nl_con'] :  bool(False)
    debug['print_ln_con'] :  bool(False)
    debug['print_objective'] :  bool(False)
```

Reply:
Also, kind of weird that you have to turn on two switches to get any debug printing, though maybe that's what Jubs wanted.  You can automatically determine the value for 'debug_print' as a logical 'or' on the other ones.

Reply:
why not just have one option (self.options['debug_print'])  that defaults to an empty list, and you can set it to ['desvars', 'nl_con', ..]?  Seems like less work than setting all of those individual bools.

Reply:
agreed that we shouldn't have two swtiches... I was thinking that wed just turn on 'debug_print` and it would split out everything. 

I likes brets suggestion, but I don't think our options_dict currently allows that kind of multi_enum input does it?

Reply:
Sounds good to me with having a list. So just a list of strings?

Reply:
a list of strings, but you have to enforce which ones are allowed

Reply:
yeah a list of strings

Reply:
OK. Looks like I can use the is_valid option to do the enforcing so that works

Reply:
valid strings ['desvars','nl_cons','ln_cons','objs']

Look good?